controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```diff
@@ -298,11 +316,7 @@ func (r *Reconciler) checkIsSourceValid(ctx *context.VirtualMachinePublishReques
 	if err != nil {
 		ctx.Logger.Error(err, "failed to get VirtualMachine", "vm", objKey)
 		if apiErrors.IsNotFound(err) {
-			// If not found the source VM, no need to requeue to cause another reconcile. Set error to nil.
-			// Ideally this won't happen because we have a validation webhook to check source VM existence.
-			errMsg := fmt.Sprintf("%s, a new VirtualMachinePublishRequest is needed to continue publishing.", err.Error())
-			vmPubReq.MarkSourceValid(corev1.ConditionFalse, vmopv1alpha1.SourceVirtualMachineNotExistReason, errMsg)
-			err = nil
+			vmPubReq.MarkSourceValid(corev1.ConditionFalse, vmopv1alpha1.SourceVirtualMachineNotExistReason, err.Error())
```

Comment:
Yeah, I removed the source VM check from the validation webhook. In this case, users are allowed to create the source VM after the vmPub is created. (I removed this check from webhook because that there is redundancy between webhook and controller. We still have to check source VM existence in the controller since the condition may not be true during reconcile.)

Reply:
@yi0909 I think having the source VM check in validation webook can still be valuable. It helps prevent reconciliation errors and provides immediate notifications from `kubectl apply/create`, improving the user experience compared to manually checking the status of the vmpub resource post-creation.